src/services/experiment-service.js

```diff
@@ -0,0 +1,113 @@
+import { EXPERIMENT_ENDPOINT, API_TIMEOUT } from '../global-constants';
+const axios = require('axios');
+
+let RETRY_LIMIT = 10;
+
+async function _getExperimentPercent(storeURL, account, experimentID) {
+    const url = `${storeURL}${EXPERIMENT_ENDPOINT}?key=${experimentID}`;
+    try {
+        const response = await axios({
+            url: url,
+            method: 'GET',
+            timeout: API_TIMEOUT,
+            headers: { 'Accept': 'application/json', 'Content-Type': 'application/json' },
+            auth: { username: account.email_address, password: account.data.auth_token },
+        });
+
+        return response.data || 0;
+    } catch (error) {
+        /**
+         * When tried to set an experiment too soon after a store is created
+         * we might get a 509. Wait for n seconds and retry a few times
+         */
+        if (error.response.status === 509) {
+            console.log('...');
+            if (RETRY_LIMIT < 1) {
+                _logMessage(`Failed to retrieve config percent for experiment ${experimentID}`);
+                throw error;
+            }
+            await new Promise(r => setTimeout(r, 3000));
+            RETRY_LIMIT--;
+            return await _getExperimentPercent(url, account, experimentID);
+        }
+        _logMessage(`Failed to get config percent for experiment ${experimentID}`);
+        throw (error);
+    }
+}
+
+async function _setExperimentPercentage(url, percentage, experimentID, account) {
+    return await axios({
+        url: url,
+        method: 'POST',
+        timeout: API_TIMEOUT,
+        headers: { 'Accept': 'application/json', 'Content-Type': 'application/json' },
+        auth: { username: account.email_address, password: account.data.auth_token },
+        data: {
+            [experimentID]: percentage.toString(),
+        },
+    });
+}
+
+function _logInitialExperimentPercentage(storeURL, account, experimentID) {
+    const initialPercent = _getExperimentPercent(storeURL, account, experimentID);
+    _logMessage(`${experimentID} is initially set to ${JSON.stringify(initialPercent)}`);
+}
+
+function _logMessage(message) {
+    console.log(message);
+}
+
+/**
+ *
+ * @param storeURL - store canonical URL or hostname
+ * @param account - consists account details
+ * @param experimentID - experiment id to rampup
+ * @returns {Promise<*>}
+ */
+async function setupExperimentConfig(storeURL, account, experimentID) {
+    _logInitialExperimentPercentage(storeURL, account, experimentID);
+    const url = `${storeURL}${EXPERIMENT_ENDPOINT}?key=${experimentID}`;
```

Comment:
Its better to construct these to avoid any input errors. For example, the `URL` constructor will fix issues with double `/` in your pathing.

```suggestion
    const url = new URL(EXPERIMENT_ENDPOINT, storeUrl);;
    url.searchParams.append('key', experimentID);
```

When you want to use it, you could call `url.toString()`.

Reply:
Didn't know about this. Will update it!

Reply:
Most languages have something similar. I try to use them as much as possibly simply because it'll save your butt at some point.